bundles/binding/org.openhab.binding.gc100ir/src/main/java/org/openhab/binding/gc100ir/internal/GC100IRBinding.java

```diff
@@ -0,0 +1,261 @@
+/**
+ * Copyright (c) 2010-2016, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.gc100ir.internal;
+
+import java.util.Dictionary;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.openhab.binding.gc100ir.GC100IRBindingProvider;
+import org.openhab.binding.gc100ir.util.GC100ItemBean;
+import org.openhab.binding.gc100ir.util.GC100ItemUtility;
+import org.openhab.core.binding.AbstractBinding;
+import org.openhab.core.binding.BindingProvider;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.osgi.service.cm.ConfigurationException;
+import org.osgi.service.cm.ManagedService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This refresh service for the GC100 binding is used to periodically check to
+ * ensure all GC100 sockets are still open and alive.
+ * 
+ * All item commands are sent via the tcp socket
+ * 
+ * @author Parikshit Thakur & Team
+ * @since 1.9.0
+ * 
+ */
+public class GC100IRBinding extends AbstractBinding<GC100IRBindingProvider> implements ManagedService {
+
+	private static final Logger logger = LoggerFactory.getLogger(GC100IRBinding.class);
+
+	private Map<String, GC100IRConnector> connectors = new HashMap<String, GC100IRConnector>();
+	private Map<String, GC100IRHost> nameHostMapper = null;
+
+	/**
+	 * the refresh interval which is used to check for lost connections
+	 * (optional, defaults to 60000ms)
+	 */
+	private long refreshInterval = 60000;
+
+	/**
```

Comment:
please add the following two methods to this class (in whatever position you prefer):
```java
    protected void addBindingProvider(GC100IRBindingProvider bindingProvider) {
        super.addBindingProvider(bindingProvider);
    }

    protected void removeBindingProvider(GC100IRBindingProvider bindingProvider) {
        super.removeBindingProvider(bindingProvider);
    }
```